roadmap/implementers-guide/src/runtime/router.md

```diff
@@ -19,6 +17,91 @@ RelayDispatchQueues: map ParaId => Vec<UpwardMessage>;
 RelayDispatchQueueSize: map ParaId => (u32, u32);
 /// The ordered list of `ParaId`s that have a `RelayDispatchQueue` entry.
 NeedsDispatch: Vec<ParaId>;
+/// The downward messages addressed for a certain para.
+DownwardMessageQueues: map ParaId => Vec<DownwardMessage>;
+```
+
+### HRMP
+
+HRMP related structs:
+
+```rust,ignore
+/// A type used to designate a HRMP channel between a (sender, recipient).
+type HrmpChannelId = (ParaId, ParaId);
+
+/// A description of a request to open an HRMP channel.
+struct HrmpOpenChannelRequest {
+	/// The sender and the initiator of this request.
+    sender: ParaId,
+	/// The recipient of the opened request.
+    recipient: ParaId,
+	/// Indicates if this request was confirmed by the recipient.
+    confirmed: bool,
+	/// How many session boundaries this request has seen.
+    age: u32,
```

Comment:
Do we use `SessionIndex` type anywhere else in guide? I guess this should be it

Reply:
Oh, nice to know